### ISSUE TITLE
The `always` option of `declaration-block-trailing-semicolon` declines a semicolon that has a place in front of the comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#423]
The `always` option of `declaration-block-trailing-semicolon` has one place to write its semicolon and knows of another. Where the last node of a block ends in a comment opened by a double slash, the character cannot go behind that comment — it would be commented out, which is what [#232](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/232) closed and what `writesIntoInlineComment` has stood in front of ever since. It can go in front of the comment, and under `postcss-scss` it does:

```scss
a {
	color: pink // c
}
```

comes back from `--fix` as `a {⏎	color: pink; // c⏎}` and the warning clears. The same file under `postcss-less` is reported at `2:17` and left alone, and no run of `--fix` can ever clear it.

## The four cells

`postcss-less` swallows such a comment into the text of the node in front of it — into a declaration's value, into an at-rule's parameters — even where the author wrote it on a line of its own. `postcss-scss` hands it over beside the node instead. So the rule's answer turns on where the parser filed the comment rather than on what the file says:

| Last node of the block | `postcss-scss` | `postcss-less` |
| --- | --- | --- |
| `color: pink // c` | `color: pink; // c` | reported, declined |
| `color: pink`<br>`// c` | `color: pink;`<br>`// c` | reported, declined |
| `@extend .b // c` | reported, declined | reported, declined |
| `@extend .b`<br>`// c` | reported, declined | reported, declined |

Under `postcss-less` the node holds `value: "pink\n\t// c"` for the second row and `params: ".b\n\t// c"` for the fourth, so the comment is inside the one text the guard is asked about, and the guard answers that a write behind it lands inside a comment. It does — and the fix has never asked whether the character could stand anywhere else.

## Every one of the four writes is one Less accepts

Measured against `less@4.9.0`:

| File | as it stands | with the semicolon in front of the comment |
| --- | --- | --- |
| `a {⏎	color: pink // c⏎}` | `a {⏎  color: pink;⏎}` | `a {⏎  color: pink;⏎}` |
| `a {⏎	color: pink⏎	// c⏎}` | `a {⏎  color: pink;⏎}` | `a {⏎  color: pink;⏎}` |
| `a {⏎	@extend .b // c⏎}` | `ParseError: @extend rule is missing block or ending semi-colon` | `a {⏎  @extend .b;⏎}` |
| `a {⏎	@extend .b⏎	// c⏎}` | `ParseError: @extend rule is missing block or ending semi-colon` | `a {⏎  @extend .b;⏎}` |

Two of the four are files Less refuses as they stand, and the write repairs them. The other two mean the same before and after. None of the four writes a character into the text of a comment, so nothing of [#211](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/211) or [#232](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/232) is undone: the semicolon closes the code, and the comment keeps saying what it said.

The control is the same file with a block comment of exactly the same width. `a {⏎	@extend .b /*c*/⏎}` comes back from `--fix` as `a {⏎	@extend .b /*c*/;⏎}` under both syntaxes, warning and all cleared, because there the character behind the comment is not inside anything. The difference between the two spellings is the whole of this issue.

## What is expected

Where the text a semicolon would follow ends in a comment opened by a double slash, `always` writes the semicolon in front of that comment rather than declining. Where it ends in code, nothing changes.

`never` is untouched: it writes nothing, and the semicolon it takes away is a separate question, which [#359](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/359) holds.

## What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-less",
  "rules": {
    "@stylistic/declaration-block-trailing-semicolon": "always"
  }
}
```
